### ISSUE TITLE
Cabal sandboxing confusion

### ISSUE THREAD

**Issue by jvoigtlaender** [elm-lang/elm-platform#64]
In the build script, you currently do one `cabal sandbox init` at the beginning, then other `cabal sandbox init`s later on. You are creating two different sandboxes that way!

The first `cabal sandbox init`, in the root directory, creates a sandbox `root/.cabal-sandbox`, so its build results would go into `root/.cabal-sandbox/bin` etc.

Then, later, in subdirectories of `root` you do `cabal sandbox init --sandbox ..`. This is *not* reusing the earlier created sandbox. Instead, you are telling cabal to create a new sandbox that will use directories like `root/bin` etc.

This is possibly a reason for issue #62.

Moreover, it is possibly a reason for why you don't see the success in #63 that you expect there. That @garyLockyer and @mgold have different versions of `aeson` in their build plans indicates that your `cabal.config` version constraints do not get taken into account. The reason is possibly where you put it, and which sandboxes actually get used. In my experiments with https://www.stackage.org/lts/cabal.config, I always put the `cabal.config` file into the directory of the project currently being built. So `cabal.config` should then be in `root/elm-compiler`, in `root/elm-repl` etc., not in `root` itself. Maybe there is a way to have cabal pick it up from the `root` location, via sandbox sharing. But the current way is obviously not working. (As said above, all the project builds do not actually use the sandbox you created at the very beginning. Instead, that one is completely ignored, instead the first project build makes another new one, and that one is used by all later project builds.)

**Comment by jvoigtlaender** [comment]
You can see what I mean by looking at this:
```
Creating a new sandbox at /home/teddy/Programy/Elm-Platform/0.15.1/.cabal-sandbox
```
and this:
```
Creating a new sandbox at /home/teddy/Programy/Elm-Platform/0.15.1
```
in @TeddyDD's [gist](https://gist.github.com/TeddyDD/9aa89410966d3c72256a), from a single run of `BuildFromSource.hs` (which has the same problem).

**Comment by jvoigtlaender** [comment]
Closing this, since it's addressed by my pull request #65.
